tasks/testwasher.py

```diff
@@ -88,3 +102,39 @@ def process_module_results(self, module_results: List[ModuleTestResult]):
             print(failed_tests_string)
 
         return should_succeed
+
+    def is_known_flaky_test(self, failing_test, known_flaky_tests, known_flaky_tests_parents):
+        """
+        Check if a test is known to be flaky
+        The method should be called with the following arguments:
+        - failing_test: the test that is failing
+        - known_flaky_tests: the set of tests that are known to be flaky
+        - known_flaky_tests_parents: the set of tests that are known to be flaky with their parents
+        If a test is a parent of a test that is known to be flaky, the test should be considered flaky
+        For example:
+        - if TestEKSSuite/TestCPU is known to be flaky, TestEKSSuite/TestCPU/TestCPUUtilization should be considered flaky
+        - if TestEKSSuite/TestCPU is known to be flaky, TestEKSSuite should be considered flaky
+        - if TestEKSSuite/TestCPU is known to be flaky, TestEKSSuite/TestMemory should not be considered flaky
+
+        """
+        failing_test_parents = self.get_tests_parents([failing_test])
+        for parent in failing_test_parents:
+            if parent in known_flaky_tests:
+                return True
+        if failing_test in known_flaky_tests_parents:
+            return True
+        return False
+
+    def get_tests_parents(self, test_name_list):
+        """
+        Get the parent tests of a list of tests
+        For example with the test ["TestEKSSuite/TestCPU/TestCPUUtilization", "TestKindSuite/TestCPU"]
+        this method should return the set{"TestEKSSuite/TestCPU/TestCPUUtilization", "TestEKSSuite/TestCPU", "TestEKSSuite", "TestKindSuite/TestCPU", "TestKindSuite"}
+        """
+        test_family = set()
+        test_family.update(test_name_list)
```

Comment:
```suggestion
        test_family = set(test_name_list)
```
However, for the `known_flaky_tests_parents` we might not need this line I guess. I don't know how far it's helping performance :)